clouddriver-docker/src/main/groovy/com/netflix/spinnaker/clouddriver/docker/registry/api/v2/auth/DockerBearerTokenService.groovy

```diff
@@ -27,24 +26,53 @@ import retrofit.http.Path
 import retrofit.http.Query
 
 class DockerBearerTokenService {
-  private Map<String, TokenService> realmToService
-  private Map<String, DockerBearerToken> cachedTokens
-  public String basicAuth
-  public String basicAuthHeader
+  Map<String, TokenService> realmToService
+  Map<String, DockerBearerToken> cachedTokens
+  String username
+  String password
+  File passwordFile
 
   @Autowired
   String dockerApplicationName
 
-  DockerBearerTokenService(String username, String password) {
+  DockerBearerTokenService() {
     realmToService = new HashMap<String, TokenService>()
     cachedTokens = new HashMap<String, DockerBearerToken>()
-    if (username || password) {
-      basicAuth = new String(Base64.encoder.encode(("${username}:${password}").bytes))
-      basicAuthHeader = "Basic $basicAuth"
+  }
+
+  DockerBearerTokenService(String username, String password) {
+    this()
+    this.username = username
+    this.password = password
+  }
+
+  DockerBearerTokenService(String username, File passwordFile) {
+    this()
+    this.username = username
+    this.passwordFile = passwordFile
+  }
+
+  String getBasicAuth() {
+    if (!(username || password || passwordFile)) {
+      return null
+    }
+
+    def resolvedPassword = null
+
+    if (password) {
+      resolvedPassword = password
+    } else if (passwordFile) {
+      resolvedPassword = new BufferedReader(new FileReader(passwordFile))
```

Comment:
Is that all you need here? You don't need to call `getText()` or something similar on the `BufferedReader`?